newIDE/app/src/MainFrame/UseCapturesManager.js

```diff
@@ -0,0 +1,149 @@
+// @flow
+
+import * as React from 'react';
+import {
+  type LaunchCaptureOptions,
+  type CaptureOptions,
+} from '../ExportAndShare/PreviewLauncher.flow';
+import { createGameResourceSignedUrls } from '../Utils/GDevelopServices/Game';
+
+const useCapturesManager = ({ project }: { project: ?gdProject }) => {
+  const [
+    unverifiedGameScreenshots,
+    setUnverifiedGameScreenshots,
+  ] = React.useState<
+    Array<{|
+      projectUuid: string,
+      unverifiedPublicUrl: string,
+    |}>
+  >([]);
+
+  const createCaptureOptionsForPreview = React.useCallback(
+    async (
+      launchCaptureOptions: ?LaunchCaptureOptions
+    ): Promise<CaptureOptions> => {
+      const captureOptions: CaptureOptions = {
+        screenshots: [],
+      };
+
+      try {
+        if (launchCaptureOptions && launchCaptureOptions.screenshots.length) {
+          const screenshotOptions = launchCaptureOptions.screenshots;
+          const response = await createGameResourceSignedUrls({
+            uploadType: 'game-screenshot',
+            files: screenshotOptions.map(screenshotOption => ({
+              contentType: 'image/png',
+            })),
+          });
+          const signedUrls = response.signedUrls;
+          if (!signedUrls || signedUrls.length === 0) {
+            throw new Error('No signed url returned');
+          }
+
+          captureOptions.screenshots = screenshotOptions
+            .map((screenshotOption, index) => {
+              const signedUrlInfo = signedUrls[index];
+              if (!signedUrlInfo) {
+                return null;
+              }
+
+              return {
+                delayTimeInSeconds: screenshotOption.delayTimeInSeconds,
+                signedUrl: signedUrlInfo.signedUrl,
+                publicUrl: signedUrlInfo.publicUrl,
+              };
+            })
+            .filter(Boolean);
+        }
+      } catch (error) {
+        console.error(
+          'Error caught while creating signed URLs for game resources. Skipping.',
+          error
+        );
+      }
+
+      return captureOptions;
+    },
+    []
+  );
+
+  const onCaptureFinished = React.useCallback(
```

Comment:
I like this better! All the logic to create signed url and then fetch them is here, rather then being splitted between here and the multiple files of the preview launchers.